Lib/venv/__init__.py

```diff
@@ -133,6 +133,40 @@ def _same_path(cls, path1, path2):
         else:
             return path1 == path2
 
+    @classmethod
+    def _getpath_realpath(cls, path):
+        """Mimics getpath.realpath
+
+        It only mimics it for HAVE_READLINK.
+        There are a few differences listed here:
+        - we ensure that we have a resolvable abspath first
+          (i.e. exists and no symlink loop)
+        - we stop if a candidate does not resolve to the same file
+          (this can happen with normpath)
+        """
+        result = os.path.abspath(path)
```

Comment:
`os.path.abspath` is done first as was used before.  It does normalize the path which may end-up with an invalid path in case of symlinks (e.g. "symlink_dir/../segment").